libs/remote_storage/src/local_fs.rs

```diff
@@ -660,6 +662,22 @@ mod fs_tests {
             "Second part bytes should be returned when requested"
         );
 
+        let suffix_bytes = storage
+            .download_byte_range(&upload_target, 13, None)
+            .await?
+            .download_stream;
+        let suffix_bytes = aggregate(suffix_bytes).await?;
+        let suffix = std::str::from_utf8(&suffix_bytes)?;
+        assert_eq!(upload_name, suffix);
```

Comment:
nit: Isn't this check redundant? We already check the full range below.

Reply:
Good point, I wasn't feeling too great about my chances of avoiding an off-by-one so I didn't think much. Could remove the other, but at the same time at least right now both are sort of interesting cases, and there are no tests for `end_exclusive == None`.

There is not a huge amount of churn here, I see the naming in this file is from the earliest file based days :)

Reply:
This duplicates, but it doesn't hurt.